src/services/formatting/formatting.ts

```diff
@@ -97,11 +97,23 @@ namespace ts.formatting {
     }
 
     export function formatOnSemicolon(position: number, sourceFile: SourceFile, rulesProvider: RulesProvider, options: FormatCodeSettings): TextChange[] {
-        return formatOutermostParent(position, SyntaxKind.SemicolonToken, sourceFile, options, rulesProvider, FormattingRequestKind.FormatOnSemicolon);
+        const outermostParent = findOutermostParentWithinListLevelFromPosition(position, SyntaxKind.SemicolonToken, sourceFile);
+        return formatOutermostParent(outermostParent, sourceFile, options, rulesProvider, FormattingRequestKind.FormatOnSemicolon);
+    }
+
+    export function formatOnOpeningCurly(position: number, sourceFile: SourceFile, rulesProvider: RulesProvider, options: FormatCodeSettings): TextChange[] {
+        const openingCurly = findPrecedingTokenOfKind(position, SyntaxKind.FirstPunctuation, sourceFile);
+        const block = openingCurly && openingCurly.parent;
+        if (!(block && isBlock(block))) {

```

Comment:
Checked the claim about open-curly parents. An opening curly can be typed as part of various constructs where there is no block in the AST (eg: object literals, type-literals, class declarations, interface declarations, binding names, etc). It seems reasonable to do formatting on typing the open curly in those cases.

So if we want to do formatting for the code following the curly, we should remove the `isBlock` check. Since we would format code *after* the open curly, and that may be surprising to some users, we should make the option toggle-able in some menu in VS.